File: sdk/go/jobs/enterprise_hybrid_test.go
package jobs

import (
	"testing"

	"github.com/aethelred/sdk-go/types"
)

func TestEnterpriseHybridDefault(t *testing.T) {
	t.Run("empty ProofType defaults to HYBRID", func(t *testing.T) {
		req := SubmitRequest{
			ModelHash: "0xabc123",
			InputHash: "0xdef456",
			// ProofType deliberately left empty
		}

		// Simulate the default-filling logic from Module.Submit
		if req.ProofType == "" {
			req.ProofType = types.ProofTypeHybrid
		}

		if req.ProofType != types.ProofTypeHybrid {
			t.Errorf("expected ProofType to be %q, got %q", types.ProofTypeHybrid, req.ProofType)
		}
	})

	t.Run("explicit TEE overrides the default", func(t *testing.T) {
		req := SubmitRequest{
			ModelHash: "0xabc123",
			InputHash: "0xdef456",
			ProofType: types.ProofTypeTEE,
		}

		// Simulate the default-filling logic from Module.Submit
		if req.ProofType == "" {
			req.ProofType = types.ProofTypeHybrid
		}

		if req.ProofType != types.ProofTypeTEE {
			t.Errorf("expected ProofType to be %q, got %q", types.ProofTypeTEE, req.ProofType)
		}
	})

	t.Run("ProofTypeHybrid constant exists", func(t *testing.T) {
		if types.ProofTypeHybrid != "PROOF_TYPE_HYBRID" {
			t.Errorf("expected ProofTypeHybrid to be %q, got %q", "PROOF_TYPE_HYBRID", types.ProofTypeHybrid)
		}
	})
}

File: sdk/typescript/src/jobs/enterprise-hybrid.test.ts
/**
 * Enterprise Hybrid Default Tests
 *
 * Verifies that the SDK defaults proofType to HYBRID when not specified.
 */

import { describe, expect, it } from 'vitest';

import { ProofType } from '../core/types';

describe('Enterprise Hybrid Defaults', () => {
  it('should default proofType to HYBRID when submit is called without proofType', () => {
    // Simulate the default-filling logic from JobsModule.submit()
    const request = {
      modelHash: '0x' + 'a'.repeat(64),
      inputHash: '0x' + 'b'.repeat(64),
    };

    const withDefaults = {
      proofType: ProofType.HYBRID,
      ...request,
    };

    expect(withDefaults.proofType).toBe(ProofType.HYBRID);
    expect(withDefaults.proofType).toBe('PROOF_TYPE_HYBRID');
  });

  it('should respect explicit proofType override', () => {
    const request = {
      modelHash: '0x' + 'a'.repeat(64),
      inputHash: '0x' + 'b'.repeat(64),
      proofType: ProofType.TEE,
    };

    const withDefaults = {
      proofType: ProofType.HYBRID,
      ...request,
    };

    // Spread puts request.proofType last, overriding the default
    expect(withDefaults.proofType).toBe(ProofType.TEE);
  });

  it('should have HYBRID in the ProofType enum', () => {
    expect(ProofType.HYBRID).toBeDefined();
    expect(ProofType.HYBRID).toBe('PROOF_TYPE_HYBRID');
  });

  it('should have all expected proof types', () => {
    expect(ProofType.TEE).toBe('PROOF_TYPE_TEE');
    expect(ProofType.ZKML).toBe('PROOF_TYPE_ZKML');
    expect(ProofType.HYBRID).toBe('PROOF_TYPE_HYBRID');
    expect(ProofType.OPTIMISTIC).toBe('PROOF_TYPE_OPTIMISTIC');
  });
});
